lib/vagrant-libvirt/config.rb

```diff
@@ -317,6 +323,18 @@ def channel(options={})
         })
       end
 
+      def random(options={})
+        if !options[:model].nil? && options[:model] != "random"
+          raise 'The only supported rng backend is "random".'
+        end
+
+        if @rng == UNSET_VALUE
```

Comment:
> and function is not used.

The function is. When specifying the random device in the Vagrantfile like mentioned in the readme:

```
  config.vm.provider :libvirt do |domain|
      domain.random :model => 'random'
  end
```

it is called.

> it should be first line of method

But all other functions have exactly this structure. First input validation, then checking if the target config property is UNSET_VALUE and acting accordingly.

But maybe I miss something here ...